File: Project Files/R/data_processing.R
library(dplyr)
library(caret)

load_data <- function() {
  df <- read.csv("../Datasets/time_series_covid-19_confirmed.csv", check.names = FALSE)

  # Filter for India
  india <- df %>% filter(`Country/Region` == "India")

  # Drop non-date columns
  india <- india[, !(names(india) %in% c("Province/State", "Country/Region", "Lat", "Long"))]

  # Sum across provinces
  cases <- colSums(india)

  # Create DataFrame
  dates <- names(cases)
  df_cases <- data.frame(
    date = as.Date(dates, format = "%m/%d/%y"),
    cases = as.numeric(cases),
    days = 1:length(cases)
  )

  # Train-test split (80/20)
  set.seed(42)
  splitIndex <- createDataPartition(df_cases$cases, p = 0.8, list = FALSE)
  train <- df_cases[splitIndex, ]
  test <- df_cases[-splitIndex, ]

  list(
    X_train = train[, "days", drop = FALSE],
    y_train = train$cases,
    X_test = test[, "days", drop = FALSE],
    y_test = test$cases
  )
}

File: Project Files/Python/data_processing.py
import pandas as pd
from sklearn.model_selection import train_test_split

def load_data():
    # Load the COVID-19 dataset
    df = pd.read_csv("../../Datasets/time_series_covid-19_confirmed.csv")

    # Filter only for India
    india_df = df[df['Country/Region'] == 'India']

    # Drop unnecessary columns
    india_df = india_df.drop(['Province/State', 'Country/Region', 'Lat', 'Long'], axis=1)

    # Sum across provinces (if there are multiple entries for India)
    india_df = india_df.sum(axis=0)

    # Convert to proper DataFrame
    india_df = pd.DataFrame({
        'date': india_df.index,
        'cases': india_df.values
    })

    # Add 'days' column
    india_df['days'] = range(1, len(india_df) + 1)

    # Features and labels
    X = india_df[['days']]
    y = india_df['cases']

    # Split into training and testing
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

    return X_train, X_test, y_train, y_test